runtime/doc/lua.txt

```diff
@@ -3034,6 +2984,478 @@ vim.glob.to_lpeg({pattern})                               *vim.glob.to_lpeg()*
+          return lpeg.match(p, s)
+        end
+        local a, b, c = split('a,b,c', ',')
+        assert(a == 'a')
+        assert(b == 'b')
+        assert(c == 'c')
+<
+
+    Parameters: ~
+      • {patt}  (`vim.lpeg.Pattern`)
+
+    Return: ~
+        (`vim.lpeg.Capture`)
+
+vim.lpeg.Carg({n})                                           *vim.lpeg.Carg()*
+    Creates an argument capture. This pattern matches the empty string and
+    produces the value given as the nth extra argument given in the call to `lpeg.match` .
+
+    Parameters: ~
+      • {n}  (`integer`)
+
+    Return: ~
+        (`vim.lpeg.Capture`)
+
+vim.lpeg.Cb({name})                                            *vim.lpeg.Cb()*
+    Creates a back capture. This pattern matches the empty string and produces
+    the values produced by the most recent group capture named `name` (where
+    `name` can be any Lua value). Most recent means the last complete
+    outermost group capture with the given name. A Complete capture means that
+    the entire pattern corresponding to the capture has matched. An Outermost
+    capture means that the capture is not inside another complete capture. In
+    the same way that LPeg does not specify when it evaluates captures, it
+    does not specify whether it reuses values previously produced by the group
+    or re-evaluates them.
+
+    Parameters: ~
+      • {name}  (`any`)
+
+    Return: ~
+        (`vim.lpeg.Capture`)
+
+vim.lpeg.Cc({...})                                             *vim.lpeg.Cc()*
+    Creates a constant capture. This pattern matches the empty string and
+    produces all given values as its captured values.
+
+    Parameters: ~
+      • {...}  (`any`)
+
+    Return: ~
+        (`vim.lpeg.Capture`)
+
+vim.lpeg.Cf({patt}, {func})                                    *vim.lpeg.Cf()*
+    Creates a fold capture. If `patt` produces a list of captures C1 C2 ...
+    Cn, this capture will produce the value `func(...func(func(C1, C2),
+    C3)...,Cn)`, that is, it will fold (or accumulate, or reduce) the captures
+    from `patt` using function `func`. This capture assumes that `patt` should
+    produce at least one capture with at least one value (of any type), which
+    becomes the initial value of an accumulator. (If you need a specific
+    initial value, you may prefix a constant captureto `patt`.) For each
+    subsequent capture, LPeg calls `func` with this accumulator as the first
+    argument and all values produced by the capture as extra arguments; the
+    first result from this call becomes the new value for the accumulator. The
+    final value of the accumulator becomes the captured value.
+
+    Example: >lua
+        local number = lpeg.R("09") ^ 1 / tonumber
+        local list = numbernumber) ^ 0                                * ("," *
+        local function add(acc, newvalue) return acc + newvalue end
+        local sum = lpeg.Cf(list, add)
+        assert(sum:match("10,30,43") == 83)
```

Comment:
Nit: Other examples use single quotes. Let's try to keep that consistent.

Reply:
The documentation and code example were brought from #25311 (https://github.com/LuaCATS/lpeg/blob/master/library/lpeg.lua), so I would vote to not converting them for such externally-written docs (other than minor style and punctuation edit so that vimdoc generation doesn't go wonky).